### PR TITLE
internal/v4: add meta/id endpoint

### PR DESCRIPTION
**Pull request by rogpeppe**
As discussed. The specification has been updated appropriately.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/578/

File: internal/v4/api_test.go [hunk 1]
```diff
@@ -320,6 +320,26 @@ var metaEndpoints = []metaEndpoint{{
...
+	},
+	checkURL: "cs:utopic/category-2",
+	assertCheckData: func(c *gc.C, data interface{}) {
+		c.Assert(data, jc.DeepEquals, params.IdResponse{
+			Id: charm.MustParseReference("cs:utopic/category-2"),
+			Series: "utopic",
+			Name: "category",
+			Revision: 2,
+		})
+	},
```
Comment (urosj) [inline internal/v4/api_test.go:342]: It's not critical, but the test with user is missing.

**Comment by urosj** [comment]
:+1: With a comment on test.

File: internal/v4/api_test.go (hunk 1 above)
Comment (rogpeppe) [inline internal/v4/api_test.go:342]: Actually, it is tested with a user. The test entities include at least one entity with a user.

File: internal/v4/api.go [hunk 2]
```diff
@@ -80,6 +80,7 @@ func New(store *charmstore.Store, config charmstore.ServerParams) *Handler {
 				h.putMetaExtraInfoWithKey,
 				"extrainfo",
 			),
+			"id": h.entityHandler(h.metaId, "_id"),
```
Comment (mhilton) [inline internal/v4/api.go]: go fmt?

**Comment by mhilton** [comment]
:+1: LGTM, No QA.

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/579/

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**